.github/workflows/haskell-ci.yml

```diff
@@ -1,43 +1,45 @@
 name: CI
 
-on: 
+on:
   pull_request:
   push:
     branches: [master]
 
-
 jobs:
   stack:
-    name: stack / resolver ${{ matrix.resolver }} 
+    name: ${{ matrix.stack-yaml }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        stack: ["2.7.3"]
-        resolver: ["lts-16.2"]
-        ghc: ["8.8.3"]
+        stack:
+          - 2.7.3
+        stack-yaml:
+          - stack-8.6.5.yaml
+          - stack-8.8.3.yaml
+          - stack-8.10.7.yaml
 
     steps:
-    - uses: actions/checkout@v2
-      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+      - uses: actions/checkout@v2
+        if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
 
-    - uses: haskell/actions/setup@v1
-      name: Setup Haskell Stack
-      with:
-        ghc-version: ${{ matrix.ghc }}
-        stack-version: ${{ matrix.stack }}
+      - uses: haskell/actions/setup@v1
+        name: Setup Haskell Stack
+        with:
+          ghc-version: ${{ matrix.ghc }}
```

Comment:
You still need to define matrix.ghc for the setup to work

Reply:
I've removed this altogether and resorted to `{{ matrix.stack-yaml }}`.